truenas_installer/server/api/truenas_connect/api.py

```diff
@@ -77,7 +78,7 @@ async def tnc_registration_uri(context):
 
     query_params = {
         'version': config['truenas_version'],
-        'model': 'UNKNOWN',  # FIXME: Obviously fix this
+        'model': get_chassis_hardware().removeprefix('TRUENAS-'),
```

Comment:
It's not an async-compatible method, it does subprocessing. Let's just query this information on startup and add it to `Installer` object that can be accessed as `context.server.installer`.

Reply:
We read DMI there, so you can call `get_chassis_hardware(dmi)` to avoid querying DMI twice.